### ISSUE TITLE
[1.16.2/1.17?] ONNX build Windows

### ISSUE THREAD

**Issue by jeremyfowers** [onnx/onnx#6267]
# Bug Report

### Is the issue related to model conversion?
No. I can't even perform imports.

### Describe the bug
My projects are permissive with respect to which `onnx` PyPI package version is installed. `onnx 1.16.2` came out this morning and broke my projects. 

For example, in a turnkeyml environment that used to work:

```
pip install --upgrade onnx
turnkey -h
```

results in:

```
(tkml) PS C:\work\turnkeyml> turnkey -h 
Traceback (most recent call last):
  File "C:\Users\jefowers\AppData\Local\miniconda3\envs\tkml\lib\runpy.py", line 194, in _run_module_as_main  
    return _run_code(code, main_globals, None,
  File "C:\Users\jefowers\AppData\Local\miniconda3\envs\tkml\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\jefowers\AppData\Local\miniconda3\envs\tkml\Scripts\turnkey.exe\__main__.py", line 4, in <module>
  File "C:\work\turnkeyml\src\turnkeyml\__init__.py", line 3, in <module>
    from .files_api import evaluate_files
  File "C:\work\turnkeyml\src\turnkeyml\files_api.py", line 8, in <module>
    from turnkeyml.sequence import Sequence
  File "C:\work\turnkeyml\src\turnkeyml\sequence\__init__.py", line 1, in <module>
    from .sequence import Sequence
  File "C:\work\turnkeyml\src\turnkeyml\sequence\sequence.py", line 11, in <module>
    import turnkeyml.common.status as status
  File "C:\work\turnkeyml\src\turnkeyml\common\status.py", line 10, in <module>
    import turnkeyml.common.analyze_model as analyze_model
  File "C:\work\turnkeyml\src\turnkeyml\common\analyze_model.py", line 4, in <module>
    import onnx
  File "C:\Users\jefowers\AppData\Local\miniconda3\envs\tkml\lib\site-packages\onnx\__init__.py", line 77, in <module>
    from onnx.onnx_cpp2py_export import ONNX_ML
ImportError: DLL load failed while importing onnx_cpp2py_export: A dynamic link library (DLL) initialization routine failed.
```

Setting `onnx<1.16.2` resolves the issue.

### System information

- OS Platform and Distribution (*e.g. Linux Ubuntu 20.04*): Windows 11
- ONNX version (*e.g. 1.13*): 1.16.2
- Python version: 3.10 (works on Python 3.8)
- Protobuf version:  3.20.2

### Reproduction instructions

```
conda create -n otest python=3.10
conda activate otest
pip install turnkeyml==3.0.1
turnkey -h
```

### Expected behavior
Patch version increases to packages (1.16.1 -> 1.16.2) should not include breaking changes.

**Comment by andife** [comment]
Hi,

that's how I tried to reproduce the issue (also Windows11):
```

(base) C:\Users\andreas>conda create -n 20240801_onnxtest python=3.8
Retrieving notices: ...working... done
Channels:
 - conda-forge
Platform: win-64
Collecting package metadata (repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
    current version: 23.11.0
    latest version: 24.7.1

Please update conda by running

    $ conda update -n base -c conda-forge conda



## Package Plan ##

  environment location: C:\Users\andreas\miniforge3\envs\20240801_onnxtest

  added / updated specs:
    - python=3.8


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    pip-24.2                   |     pyhd8ed1ab_0         1.2 MB  conda-forge
    python-3.8.19              |h4de0772_0_cpython        15.3 MB  conda-forge
    ------------------------------------------------------------
                                           Total:        16.5 MB

The following NEW packages will be INSTALLED:

  bzip2              conda-forge/win-64::bzip2-1.0.8-h2466b09_7
  ca-certificates    conda-forge/win-64::ca-certificates-2024.7.4-h56e8100_0
  libffi             conda-forge/win-64::libffi-3.4.2-h8ffe710_5
  libsqlite          conda-forge/win-64::libsqlite-3.46.0-h2466b09_0
  libzlib            conda-forge/win-64::libzlib-1.3.1-h2466b09_1
  openssl            conda-forge/win-64::openssl-3.3.1-h2466b09_2
  pip                conda-forge/noarch::pip-24.2-pyhd8ed1ab_0
  python             conda-forge/win-64::python-3.8.19-h4de0772_0_cpython
  setuptools         conda-forge/noarch::setuptools-71.0.4-pyhd8ed1ab_0
  tk                 conda-forge/win-64::tk-8.6.13-h5226925_1
  ucrt               conda-forge/win-64::ucrt-10.0.22621.0-h57928b3_0
  vc                 conda-forge/win-64::vc-14.3-h8a93ad2_20
  vc14_runtime       conda-forge/win-64::vc14_runtime-14.40.33810-ha82c5b3_20
  vs2015_runtime     conda-forge/win-64::vs2015_runtime-14.40.33810-h3bf8584_20
  wheel              conda-forge/noarch::wheel-0.43.0-pyhd8ed1ab_1
  xz                 conda-forge/win-64::xz-5.2.6-h8d14728_0


Proceed ([y]/n)? y


Downloading and Extracting Packages:

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate 20240801_onnxtest
#
# To deactivate an active environment, use
#
#     $ conda deactivate


(base) C:\Users\andreas>conda activate 20240801_onnxtest

(20240801_onnxtest) C:\Users\andreas>pip install --upgrade onnx
Collecting onnx
  Downloading onnx-1.16.2-cp38-cp38-win_amd64.whl.metadata (16 kB)
Collecting numpy>=1.20 (from onnx)
  Downloading numpy-1.24.4-cp38-cp38-win_amd64.whl.metadata (5.6 kB)
Collecting protobuf>=3.20.2 (from onnx)
  Downloading protobuf-5.27.3-cp38-cp38-win_amd64.whl.metadata (592 bytes)
Downloading onnx-1.16.2-cp38-cp38-win_amd64.whl (14.4 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 14.4/14.4 MB 6.8 MB/s eta 0:00:00
Downloading numpy-1.24.4-cp38-cp38-win_amd64.whl (14.9 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 14.9/14.9 MB 6.8 MB/s eta 0:00:00
Downloading protobuf-5.27.3-cp38-cp38-win_amd64.whl (426 kB)
Installing collected packages: protobuf, numpy, onnx
Successfully installed numpy-1.24.4 onnx-1.16.2 protobuf-5.27.3

(20240801_onnxtest) C:\Users\andreas>pip install turnkey
ERROR: Could not find a version that satisfies the requirement turnkey (from versions: none)
ERROR: No matching distribution found for turnkey

(20240801_onnxtest) C:\Users\andreas>pip install turnkeyml
Collecting turnkeyml
  Downloading turnkeyml-3.0.2-py3-none-any.whl.metadata (9.7 kB)
Collecting invoke>=2.0.0 (from turnkeyml)
  Downloading invoke-2.2.0-py3-none-any.whl.metadata (3.3 kB)
Collecting onnx<1.16.2,>=1.11.0 (from turnkeyml)
  Downloading onnx-1.16.1-cp38-cp38-win_amd64.whl.metadata (16 kB)
Collecting onnxmltools==1.10.0 (from turnkeyml)
  Downloading onnxmltools-1.10.0-py2.py3-none-any.whl.metadata (8.3 kB)
Collecting onnxruntime>=1.10.1 (from turnkeyml)
  Downloading onnxruntime-1.18.1-cp38-cp38-win_amd64.whl.metadata (4.5 kB)
Collecting torch>=1.12.1 (from turnkeyml)
  Downloading torch-2.4.0-cp38-cp38-win_amd64.whl.metadata (27 kB)
Collecting protobuf<3.21,>=3.17.3 (from turnkeyml)
  Downloading protobuf-3.20.3-cp38-cp38-win_amd64.whl.metadata (699 bytes)
Collecting pyyaml>=5.4 (from turnkeyml)
  Downloading PyYAML-6.0.1-cp38-cp38-win_amd64.whl.metadata (2.1 kB)
Collecting typeguard>=2.3.13 (from turnkeyml)
  Downloading typeguard-4.3.0-py3-none-any.whl.metadata (3.7 kB)
Collecting packaging>=20.9 (from turnkeyml)
  Using cached packaging-24.1-py3-none-any.whl.metadata (3.2 kB)
Requirement already satisfied: numpy<2.0.0 in c:\users\andreas\miniforge3\envs\20240801_onnxtest\lib\site-packages (from turnkeyml) (1.24.4)
Collecting pandas>=1.5.3 (from turnkeyml)
  Downloading pandas-2.0.3-cp38-cp38-win_amd64.whl.metadata (18 kB)
Collecting fasteners (from turnkeyml)
  Downloading fasteners-0.19-py3-none-any.whl.metadata (4.9 kB)
Collecting GitPython>=3.1.40 (from turnkeyml)
  Downloading GitPython-3.1.43-py3-none-any.whl.metadata (13 kB)
Collecting psutil (from turnkeyml)
  Downloading psutil-6.0.0-cp37-abi3-win_amd64.whl.metadata (22 kB)
Collecting skl2onnx (from onnxmltools==1.10.0->turnkeyml)
  Downloading skl2onnx-1.17.0-py2.py3-none-any.whl.metadata (3.2 kB)
Collecting gitdb<5,>=4.0.1 (from GitPython>=3.1.40->turnkeyml)
  Downloading gitdb-4.0.11-py3-none-any.whl.metadata (1.2 kB)
Collecting coloredlogs (from onnxruntime>=1.10.1->turnkeyml)
  Downloading coloredlogs-15.0.1-py2.py3-none-any.whl.metadata (12 kB)
Collecting flatbuffers (from onnxruntime>=1.10.1->turnkeyml)
  Using cached flatbuffers-24.3.25-py2.py3-none-any.whl.metadata (850 bytes)
Collecting sympy (from onnxruntime>=1.10.1->turnkeyml)
  Using cached sympy-1.13.1-py3-none-any.whl.metadata (12 kB)
Collecting python-dateutil>=2.8.2 (from pandas>=1.5.3->turnkeyml)
  Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting pytz>=2020.1 (from pandas>=1.5.3->turnkeyml)
  Using cached pytz-2024.1-py2.py3-none-any.whl.metadata (22 kB)
Collecting tzdata>=2022.1 (from pandas>=1.5.3->turnkeyml)
  Downloading tzdata-2024.1-py2.py3-none-any.whl.metadata (1.4 kB)
Collecting filelock (from torch>=1.12.1->turnkeyml)
  Using cached filelock-3.15.4-py3-none-any.whl.metadata (2.9 kB)
Collecting typing-extensions>=4.8.0 (from torch>=1.12.1->turnkeyml)
  Using cached typing_extensions-4.12.2-py3-none-any.whl.metadata (3.0 kB)
Collecting networkx (from torch>=1.12.1->turnkeyml)
  Downloading networkx-3.1-py3-none-any.whl.metadata (5.3 kB)
Collecting jinja2 (from torch>=1.12.1->turnkeyml)
  Using cached jinja2-3.1.4-py3-none-any.whl.metadata (2.6 kB)
Collecting fsspec (from torch>=1.12.1->turnkeyml)
  Using cached fsspec-2024.6.1-py3-none-any.whl.metadata (11 kB)
Collecting importlib-metadata>=3.6 (from typeguard>=2.3.13->turnkeyml)
  Downloading importlib_metadata-8.2.0-py3-none-any.whl.metadata (4.7 kB)
Collecting smmap<6,>=3.0.1 (from gitdb<5,>=4.0.1->GitPython>=3.1.40->turnkeyml)
  Downloading smmap-5.0.1-py3-none-any.whl.metadata (4.3 kB)
Collecting zipp>=0.5 (from importlib-metadata>=3.6->typeguard>=2.3.13->turnkeyml)
  Using cached zipp-3.19.2-py3-none-any.whl.metadata (3.6 kB)
Collecting six>=1.5 (from python-dateutil>=2.8.2->pandas>=1.5.3->turnkeyml)
  Using cached six-1.16.0-py2.py3-none-any.whl.metadata (1.8 kB)
Collecting humanfriendly>=9.1 (from coloredlogs->onnxruntime>=1.10.1->turnkeyml)
  Downloading humanfriendly-10.0-py2.py3-none-any.whl.metadata (9.2 kB)
Collecting MarkupSafe>=2.0 (from jinja2->torch>=1.12.1->turnkeyml)
  Downloading MarkupSafe-2.1.5-cp38-cp38-win_amd64.whl.metadata (3.1 kB)
Collecting scikit-learn>=1.1 (from skl2onnx->onnxmltools==1.10.0->turnkeyml)
  Downloading scikit_learn-1.3.2-cp38-cp38-win_amd64.whl.metadata (11 kB)
Collecting onnxconverter-common>=1.7.0 (from skl2onnx->onnxmltools==1.10.0->turnkeyml)
  Downloading onnxconverter_common-1.14.0-py2.py3-none-any.whl.metadata (4.2 kB)
Collecting mpmath<1.4,>=1.1.0 (from sympy->onnxruntime>=1.10.1->turnkeyml)
  Using cached mpmath-1.3.0-py3-none-any.whl.metadata (8.6 kB)
Collecting pyreadline3 (from humanfriendly>=9.1->coloredlogs->onnxruntime>=1.10.1->turnkeyml)
  Downloading pyreadline3-3.4.1-py3-none-any.whl.metadata (2.0 kB)
Collecting protobuf<3.21,>=3.17.3 (from turnkeyml)
  Downloading protobuf-3.20.2-cp38-cp38-win_amd64.whl.metadata (699 bytes)
Collecting scipy>=1.5.0 (from scikit-learn>=1.1->skl2onnx->onnxmltools==1.10.0->turnkeyml)
  Downloading scipy-1.10.1-cp38-cp38-win_amd64.whl.metadata (58 kB)
Collecting joblib>=1.1.1 (from scikit-learn>=1.1->skl2onnx->onnxmltools==1.10.0->turnkeyml)
  Using cached joblib-1.4.2-py3-none-any.whl.metadata (5.4 kB)
Collecting threadpoolctl>=2.0.0 (from scikit-learn>=1.1->skl2onnx->onnxmltools==1.10.0->turnkeyml)
  Using cached threadpoolctl-3.5.0-py3-none-any.whl.metadata (13 kB)
Downloading turnkeyml-3.0.2-py3-none-any.whl (805 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 805.1/805.1 kB 5.0 MB/s eta 0:00:00
Downloading onnxmltools-1.10.0-py2.py3-none-any.whl (300 kB)
Downloading GitPython-3.1.43-py3-none-any.whl (207 kB)
Downloading invoke-2.2.0-py3-none-any.whl (160 kB)
Downloading onnx-1.16.1-cp38-cp38-win_amd64.whl (14.4 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 14.4/14.4 MB 6.8 MB/s eta 0:00:00
Downloading onnxruntime-1.18.1-cp38-cp38-win_amd64.whl (5.6 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.6/5.6 MB 6.8 MB/s eta 0:00:00
Using cached packaging-24.1-py3-none-any.whl (53 kB)
Downloading pandas-2.0.3-cp38-cp38-win_amd64.whl (10.8 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.8/10.8 MB 6.9 MB/s eta 0:00:00
Downloading PyYAML-6.0.1-cp38-cp38-win_amd64.whl (157 kB)
Downloading torch-2.4.0-cp38-cp38-win_amd64.whl (198.1 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╸━━━━━━━━━ 152.0/198.1 MB 6.7 MB/s eta 0:00:07
ERROR: Operation cancelled by user

(20240801_onnxtest) C:\Users\andreas>pip install turnkeyml==3.0.1
Collecting turnkeyml==3.0.1
  Downloading turnkeyml-3.0.1-py3-none-any.whl.metadata (9.7 kB)
Collecting invoke>=2.0.0 (from turnkeyml==3.0.1)
  Using cached invoke-2.2.0-py3-none-any.whl.metadata (3.3 kB)
Requirement already satisfied: onnx>=1.11.0 in c:\users\andreas\miniforge3\envs\20240801_onnxtest\lib\site-packages (from turnkeyml==3.0.1) (1.16.2)
Collecting onnxmltools==1.10.0 (from turnkeyml==3.0.1)
  Using cached onnxmltools-1.10.0-py2.py3-none-any.whl.metadata (8.3 kB)
Collecting onnxruntime>=1.10.1 (from turnkeyml==3.0.1)
  Using cached onnxruntime-1.18.1-cp38-cp38-win_amd64.whl.metadata (4.5 kB)
Collecting torch>=1.12.1 (from turnkeyml==3.0.1)
  Using cached torch-2.4.0-cp38-cp38-win_amd64.whl.metadata (27 kB)
Collecting protobuf<3.21,>=3.17.3 (from turnkeyml==3.0.1)
  Using cached protobuf-3.20.3-cp38-cp38-win_amd64.whl.metadata (699 bytes)
Collecting pyyaml>=5.4 (from turnkeyml==3.0.1)
  Using cached PyYAML-6.0.1-cp38-cp38-win_amd64.whl.metadata (2.1 kB)
Collecting typeguard>=2.3.13 (from turnkeyml==3.0.1)
  Using cached typeguard-4.3.0-py3-none-any.whl.metadata (3.7 kB)
Collecting packaging>=20.9 (from turnkeyml==3.0.1)
  Using cached packaging-24.1-py3-none-any.whl.metadata (3.2 kB)
Requirement already satisfied: numpy<2.0.0 in c:\users\andreas\miniforge3\envs\20240801_onnxtest\lib\site-packages (from turnkeyml==3.0.1) (1.24.4)
Collecting pandas>=1.5.3 (from turnkeyml==3.0.1)
  Using cached pandas-2.0.3-cp38-cp38-win_amd64.whl.metadata (18 kB)
Collecting fasteners (from turnkeyml==3.0.1)
  Using cached fasteners-0.19-py3-none-any.whl.metadata (4.9 kB)
Collecting GitPython>=3.1.40 (from turnkeyml==3.0.1)
  Using cached GitPython-3.1.43-py3-none-any.whl.metadata (13 kB)
Collecting psutil (from turnkeyml==3.0.1)
  Using cached psutil-6.0.0-cp37-abi3-win_amd64.whl.metadata (22 kB)
Collecting skl2onnx (from onnxmltools==1.10.0->turnkeyml==3.0.1)
  Using cached skl2onnx-1.17.0-py2.py3-none-any.whl.metadata (3.2 kB)
Collecting gitdb<5,>=4.0.1 (from GitPython>=3.1.40->turnkeyml==3.0.1)
  Using cached gitdb-4.0.11-py3-none-any.whl.metadata (1.2 kB)
Collecting coloredlogs (from onnxruntime>=1.10.1->turnkeyml==3.0.1)
  Using cached coloredlogs-15.0.1-py2.py3-none-any.whl.metadata (12 kB)
Collecting flatbuffers (from onnxruntime>=1.10.1->turnkeyml==3.0.1)
  Using cached flatbuffers-24.3.25-py2.py3-none-any.whl.metadata (850 bytes)
Collecting sympy (from onnxruntime>=1.10.1->turnkeyml==3.0.1)
  Using cached sympy-1.13.1-py3-none-any.whl.metadata (12 kB)
Collecting python-dateutil>=2.8.2 (from pandas>=1.5.3->turnkeyml==3.0.1)
  Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting pytz>=2020.1 (from pandas>=1.5.3->turnkeyml==3.0.1)
  Using cached pytz-2024.1-py2.py3-none-any.whl.metadata (22 kB)
Collecting tzdata>=2022.1 (from pandas>=1.5.3->turnkeyml==3.0.1)
  Using cached tzdata-2024.1-py2.py3-none-any.whl.metadata (1.4 kB)
Collecting filelock (from torch>=1.12.1->turnkeyml==3.0.1)
  Using cached filelock-3.15.4-py3-none-any.whl.metadata (2.9 kB)
Collecting typing-extensions>=4.8.0 (from torch>=1.12.1->turnkeyml==3.0.1)
  Using cached typing_extensions-4.12.2-py3-none-any.whl.metadata (3.0 kB)
Collecting networkx (from torch>=1.12.1->turnkeyml==3.0.1)
  Using cached networkx-3.1-py3-none-any.whl.metadata (5.3 kB)
Collecting jinja2 (from torch>=1.12.1->turnkeyml==3.0.1)
  Using cached jinja2-3.1.4-py3-none-any.whl.metadata (2.6 kB)
Collecting fsspec (from torch>=1.12.1->turnkeyml==3.0.1)
  Using cached fsspec-2024.6.1-py3-none-any.whl.metadata (11 kB)
Collecting importlib-metadata>=3.6 (from typeguard>=2.3.13->turnkeyml==3.0.1)
  Using cached importlib_metadata-8.2.0-py3-none-any.whl.metadata (4.7 kB)
Collecting smmap<6,>=3.0.1 (from gitdb<5,>=4.0.1->GitPython>=3.1.40->turnkeyml==3.0.1)
  Using cached smmap-5.0.1-py3-none-any.whl.metadata (4.3 kB)
Collecting zipp>=0.5 (from importlib-metadata>=3.6->typeguard>=2.3.13->turnkeyml==3.0.1)
  Using cached zipp-3.19.2-py3-none-any.whl.metadata (3.6 kB)
Collecting six>=1.5 (from python-dateutil>=2.8.2->pandas>=1.5.3->turnkeyml==3.0.1)
  Using cached six-1.16.0-py2.py3-none-any.whl.metadata (1.8 kB)
Collecting humanfriendly>=9.1 (from coloredlogs->onnxruntime>=1.10.1->turnkeyml==3.0.1)
  Using cached humanfriendly-10.0-py2.py3-none-any.whl.metadata (9.2 kB)
Collecting MarkupSafe>=2.0 (from jinja2->torch>=1.12.1->turnkeyml==3.0.1)
  Using cached MarkupSafe-2.1.5-cp38-cp38-win_amd64.whl.metadata (3.1 kB)
Collecting scikit-learn>=1.1 (from skl2onnx->onnxmltools==1.10.0->turnkeyml==3.0.1)
  Using cached scikit_learn-1.3.2-cp38-cp38-win_amd64.whl.metadata (11 kB)
Collecting onnxconverter-common>=1.7.0 (from skl2onnx->onnxmltools==1.10.0->turnkeyml==3.0.1)
  Using cached onnxconverter_common-1.14.0-py2.py3-none-any.whl.metadata (4.2 kB)
Collecting mpmath<1.4,>=1.1.0 (from sympy->onnxruntime>=1.10.1->turnkeyml==3.0.1)
  Using cached mpmath-1.3.0-py3-none-any.whl.metadata (8.6 kB)
Collecting pyreadline3 (from humanfriendly>=9.1->coloredlogs->onnxruntime>=1.10.1->turnkeyml==3.0.1)
  Using cached pyreadline3-3.4.1-py3-none-any.whl.metadata (2.0 kB)
Collecting protobuf<3.21,>=3.17.3 (from turnkeyml==3.0.1)
  Using cached protobuf-3.20.2-cp38-cp38-win_amd64.whl.metadata (699 bytes)
Collecting scipy>=1.5.0 (from scikit-learn>=1.1->skl2onnx->onnxmltools==1.10.0->turnkeyml==3.0.1)
  Using cached scipy-1.10.1-cp38-cp38-win_amd64.whl.metadata (58 kB)
Collecting joblib>=1.1.1 (from scikit-learn>=1.1->skl2onnx->onnxmltools==1.10.0->turnkeyml==3.0.1)
  Using cached joblib-1.4.2-py3-none-any.whl.metadata (5.4 kB)
Collecting threadpoolctl>=2.0.0 (from scikit-learn>=1.1->skl2onnx->onnxmltools==1.10.0->turnkeyml==3.0.1)
  Using cached threadpoolctl-3.5.0-py3-none-any.whl.metadata (13 kB)
Downloading turnkeyml-3.0.1-py3-none-any.whl (805 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 805.1/805.1 kB 7.0 MB/s eta 0:00:00
Using cached onnxmltools-1.10.0-py2.py3-none-any.whl (300 kB)
Using cached GitPython-3.1.43-py3-none-any.whl (207 kB)
Using cached invoke-2.2.0-py3-none-any.whl (160 kB)
Using cached onnxruntime-1.18.1-cp38-cp38-win_amd64.whl (5.6 MB)
Using cached packaging-24.1-py3-none-any.whl (53 kB)
Using cached pandas-2.0.3-cp38-cp38-win_amd64.whl (10.8 MB)
Using cached PyYAML-6.0.1-cp38-cp38-win_amd64.whl (157 kB)
Downloading torch-2.4.0-cp38-cp38-win_amd64.whl (198.1 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 198.1/198.1 MB 6.9 MB/s eta 0:00:00
Downloading typeguard-4.3.0-py3-none-any.whl (35 kB)
Downloading fasteners-0.19-py3-none-any.whl (18 kB)
Downloading psutil-6.0.0-cp37-abi3-win_amd64.whl (257 kB)
Downloading gitdb-4.0.11-py3-none-any.whl (62 kB)
Downloading importlib_metadata-8.2.0-py3-none-any.whl (25 kB)
Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
Using cached pytz-2024.1-py2.py3-none-any.whl (505 kB)
Using cached typing_extensions-4.12.2-py3-none-any.whl (37 kB)
Downloading tzdata-2024.1-py2.py3-none-any.whl (345 kB)
Downloading coloredlogs-15.0.1-py2.py3-none-any.whl (46 kB)
Using cached filelock-3.15.4-py3-none-any.whl (16 kB)
Using cached flatbuffers-24.3.25-py2.py3-none-any.whl (26 kB)
Using cached fsspec-2024.6.1-py3-none-any.whl (177 kB)
Using cached jinja2-3.1.4-py3-none-any.whl (133 kB)
Downloading networkx-3.1-py3-none-any.whl (2.1 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.1/2.1 MB 6.8 MB/s eta 0:00:00
Downloading skl2onnx-1.17.0-py2.py3-none-any.whl (298 kB)
Using cached sympy-1.13.1-py3-none-any.whl (6.2 MB)
Downloading humanfriendly-10.0-py2.py3-none-any.whl (86 kB)
Downloading MarkupSafe-2.1.5-cp38-cp38-win_amd64.whl (17 kB)
Using cached mpmath-1.3.0-py3-none-any.whl (536 kB)
Downloading onnxconverter_common-1.14.0-py2.py3-none-any.whl (84 kB)
Downloading protobuf-3.20.2-cp38-cp38-win_amd64.whl (904 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 904.4/904.4 kB 6.8 MB/s eta 0:00:00
Downloading scikit_learn-1.3.2-cp38-cp38-win_amd64.whl (9.3 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 9.3/9.3 MB 6.8 MB/s eta 0:00:00
Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Downloading smmap-5.0.1-py3-none-any.whl (24 kB)
Using cached zipp-3.19.2-py3-none-any.whl (9.0 kB)
Using cached joblib-1.4.2-py3-none-any.whl (301 kB)
Downloading scipy-1.10.1-cp38-cp38-win_amd64.whl (42.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.2/42.2 MB 6.8 MB/s eta 0:00:00
Using cached threadpoolctl-3.5.0-py3-none-any.whl (18 kB)
Downloading pyreadline3-3.4.1-py3-none-any.whl (95 kB)
Installing collected packages: pytz, pyreadline3, mpmath, flatbuffers, zipp, tzdata, typing-extensions, threadpoolctl, sympy, smmap, six, scipy, pyyaml, psutil, protobuf, packaging, networkx, MarkupSafe, joblib, invoke, humanfriendly, fsspec, filelock, fasteners, scikit-learn, python-dateutil, jinja2, importlib-metadata, gitdb, coloredlogs, typeguard, torch, pandas, onnxruntime, onnxconverter-common, GitPython, skl2onnx, onnxmltools, turnkeyml
  Attempting uninstall: protobuf
    Found existing installation: protobuf 5.27.3
    Uninstalling protobuf-5.27.3:
      Successfully uninstalled protobuf-5.27.3
Successfully installed GitPython-3.1.43 MarkupSafe-2.1.5 coloredlogs-15.0.1 fasteners-0.19 filelock-3.15.4 flatbuffers-24.3.25 fsspec-2024.6.1 gitdb-4.0.11 humanfriendly-10.0 importlib-metadata-8.2.0 invoke-2.2.0 jinja2-3.1.4 joblib-1.4.2 mpmath-1.3.0 networkx-3.1 onnxconverter-common-1.14.0 onnxmltools-1.10.0 onnxruntime-1.18.1 packaging-24.1 pandas-2.0.3 protobuf-3.20.2 psutil-6.0.0 pyreadline3-3.4.1 python-dateutil-2.9.0.post0 pytz-2024.1 pyyaml-6.0.1 scikit-learn-1.3.2 scipy-1.10.1 six-1.16.0 skl2onnx-1.17.0 smmap-5.0.1 sympy-1.13.1 threadpoolctl-3.5.0 torch-2.4.0 turnkeyml-3.0.1 typeguard-4.3.0 typing-extensions-4.12.2 tzdata-2024.1 zipp-3.19.2

(20240801_onnxtest) C:\Users\andreas>
(20240801_onnxtest) C:\Users\andreas>
(20240801_onnxtest) C:\Users\andreas>turnkey
usage: turnkey [-h] [- [tool --tool-args [tool --tool-args...]]] [-i INPUT_FILES [INPUT_FILES ...]] [-d CACHE_DIR]
               [--lean-cache] [--labels [LABELS [LABELS ...]]] [--use-slurm | --process-isolation] [--timeout TIMEOUT]

This utility runs tools in a sequence. To use it, provide a list of tools and their arguments. See https://github.com/onnx/turnkeyml/blob/main/docs/tools_user_guide.md to learn the exact syntax.

Example: turnkey -i my_model.py discover export-pytorch

positional arguments:
  - [tool --tool-args [tool --tool-args...]], - [tool --tool-args [tool --tool-args...]], t [tool --tool-args [tool --tool-args...]], o [tool --tool-args [tool --tool-args...]], o [tool --tool-args [tool --tool-args...]], l [tool --tool-args [tool --tool-args...]], s [tool --tool-args [tool --tool-args...]]
                        Available tools that can be sequenced together to perform a build.

                        Call `turnkey TOOL -h` to learn more about each tool.

                        Tools that can start a sequence:
                         * discover: Discover the PyTorch models in a python script
                         * load-onnx: Load an ONNX model
                         * load-build: Load build state from the cache

                        Tools that go into a sequence:
                         * benchmark: Benchmark a model
                         * export-pytorch: Export a PyTorch model to ONNX
                         * optimize-ort: Use OnnxRuntime to optimize an ONNX model
                         * convert-fp16: Use OnnxMLTools to convert an ONNX model to fp16
                         * verify-exporter: Verify if model can be exported to ONNX without major numerical
                         discrepancies

                        Management tool choices:
                         * version: Print the turnkeyml version number
                         * cache: Manage the turnkey build cache
                         * models-location: Print the location of the built-in turnkeyml models
                         * report: Export statistics from each turnkey run to a CSV file

optional arguments:
  -h, --help            show this help message and exit
  -i INPUT_FILES [INPUT_FILES ...], --input-files INPUT_FILES [INPUT_FILES ...]
                        One or more inputs that will be evaluated by the tool sequence (e.g., script (.py), ONNX
                        (.onnx), turnkey build state (state.yaml), input list (.txt) files)
  -d CACHE_DIR, --cache-dir CACHE_DIR
                        Build cache directory where results will be stored (defaults to
                        C:\Users\andreas/.cache/turnkey)
  --lean-cache          Delete all build artifacts (e.g., .onnx files) when the command completes
  --labels [LABELS [LABELS ...]]
                        Filter the --input-files to only include files that have the provided labels
  --use-slurm           Execute on Slurm instead of using local compute resources
  --process-isolation   Isolate evaluating each input into a separate process
  --timeout TIMEOUT     Build timeout, in seconds, after which a build will be canceled (default=3600). Only applies
                        when --process-isolation or --use-slurm is also used.

(20240801_onnxtest) C:\Users\andreas>pip list
Package              Version
-------------------- -----------
coloredlogs          15.0.1
fasteners            0.19
filelock             3.15.4
flatbuffers          24.3.25
fsspec               2024.6.1
gitdb                4.0.11
GitPython            3.1.43
humanfriendly        10.0
importlib_metadata   8.2.0
invoke               2.2.0
Jinja2               3.1.4
joblib               1.4.2
MarkupSafe           2.1.5
mpmath               1.3.0
networkx             3.1
numpy                1.24.4
onnx                 1.16.2
onnxconverter-common 1.14.0
onnxmltools          1.10.0
onnxruntime          1.18.1
packaging            24.1
pandas               2.0.3
pip                  24.2
protobuf             3.20.2
psutil               6.0.0
pyreadline3          3.4.1
python-dateutil      2.9.0.post0
pytz                 2024.1
PyYAML               6.0.1
scikit-learn         1.3.2
scipy                1.10.1
setuptools           71.0.4
six                  1.16.0
skl2onnx             1.17.0
smmap                5.0.1
sympy                1.13.1
threadpoolctl        3.5.0
torch                2.4.0
turnkeyml            3.0.1
typeguard            4.3.0
typing_extensions    4.12.2
tzdata               2024.1
wheel                0.43.0
zipp                 3.19.2

(20240801_onnxtest) C:\Users\andreas>
```

**Comment by jeremyfowers** [comment]
Hi @andife you are right, the problem doesn't exhibit on Python 3.8. It does exhibit on Python 3.10 though. I will update the description.

My reproduction:

```
conda create -n otest python=3.10
conda activate otest
pip install turnkeyml==3.0.1
turnkey -h
```

Results in

```
(otest) PS C:\work> turnkey -h
Traceback (most recent call last):
  File "C:\Users\jefowers\AppData\Local\miniconda3\envs\otest\lib\runpy.py", line 196, in _run_module_as_main 
    return _run_code(code, main_globals, None,
  File "C:\Users\jefowers\AppData\Local\miniconda3\envs\otest\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\jefowers\AppData\Local\miniconda3\envs\otest\Scripts\turnkey.exe\__main__.py", line 4, in <module>
  File "C:\Users\jefowers\AppData\Local\miniconda3\envs\otest\lib\site-packages\turnkeyml\__init__.py", line 3, in <module>
    from .files_api import evaluate_files
  File "C:\Users\jefowers\AppData\Local\miniconda3\envs\otest\lib\site-packages\turnkeyml\files_api.py", line 8, in <module>
    from turnkeyml.sequence import Sequence
  File "C:\Users\jefowers\AppData\Local\miniconda3\envs\otest\lib\site-packages\turnkeyml\sequence\__init__.py", line 1, in <module>
    from .sequence import Sequence
  File "C:\Users\jefowers\AppData\Local\miniconda3\envs\otest\lib\site-packages\turnkeyml\sequence\sequence.py", line 11, in <module>
    import turnkeyml.common.status as status
  File "C:\Users\jefowers\AppData\Local\miniconda3\envs\otest\lib\site-packages\turnkeyml\common\status.py", line 10, in <module>
    import turnkeyml.common.analyze_model as analyze_model
  File "C:\Users\jefowers\AppData\Local\miniconda3\envs\otest\lib\site-packages\turnkeyml\common\analyze_model.py", line 4, in <module>
    import onnx
  File "C:\Users\jefowers\AppData\Local\miniconda3\envs\otest\lib\site-packages\onnx\__init__.py", line 77, in <module>
    from onnx.onnx_cpp2py_export import ONNX_ML
ImportError: DLL load failed while importing onnx_cpp2py_export: A dynamic link library (DLL) initialization routine failed.
```

**Comment by ramkrishna2910** [comment]
@cjvolzka for visibility

**Comment by tomid99999** [comment]
Downgrading to 1.16.1 seems to fix it for me.

**Comment by Jellie122** [comment]
It happened in python=3.8.19,onnx=1.16.2 too ,i downgraded the version of oxxn(1.16.1),the problem was solved. And thx for ur sharing.

**Comment by Jellie122** [comment]
It happened in python=3.8.19,onnx=1.16.2 too ,i downgraded the version of onnx(1.16.1),the problem was solved. And thx for ur sharing.

**Comment by titaiwangms** [comment]
cc @liqunfu

**Comment by ksaur** [comment]
This happens for us in in windows in 3.9 and 3.10 an we are blocked. For now we pin `1.16.1`.

**Comment by 0utsider0** [comment]
Hello im having this same problem with 1.16.2 version and my informatics knowledge is limited, can someone tell if its possible to go on another version of onnx to make it work?

**Comment by Jellie122** [comment]
> Hello im having this same problem with 1.16.2 version and my informatics knowledge is limited, can someone tell if its possible to go on another version of onnx to make it work?
> 
> ![failed8](https://private-user-images.githubusercontent.com/176232479/354851760-70ba475b-7aba-42a9-9a90-16c64f3fa5ce.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI4MTc0NzUsIm5iZiI6MTcyMjgxNzE3NSwicGF0aCI6Ii8xNzYyMzI0NzkvMzU0ODUxNzYwLTcwYmE0NzViLTdhYmEtNDJhOS05YTkwLTE2YzY0ZjNmYTVjZS5QTkc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwNVQwMDE5MzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yMGZiMjJhYWRiM2JlMmFlYTI0MTA3MmQ5ZDNjMzBlN2E2NmUzOGQ5NmMzNjIxMDMxMzU3ZmM2ODNkNGNiMDhmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.IYu_z7SNWcpFe3snHq_NcWrva61BtPvWUJQhOBrbvpg)
> 
> Can someone please explain me why you are talking about 1.16.2 version and I have 1.18?
> 
> When I try tu uninstall onnxruntime it says I have 1.18 version, but when I try to upgrade it says requirement already satisfied and I have 1.16.2???? ![failed9](https://private-user-images.githubusercontent.com/176232479/354852047-c53e7b08-67e4-4c8d-bc3f-f001b40909b9.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI4MTc0NzUsIm5iZiI6MTcyMjgxNzE3NSwicGF0aCI6Ii8xNzYyMzI0NzkvMzU0ODUyMDQ3LWM1M2U3YjA4LTY3ZTQtNGM4ZC1iYzNmLWYwMDFiNDA5MDliOS5QTkc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODA1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwNVQwMDE5MzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00Y2ZiMGVkMjBkNTVjNDk1NTlkYjkxYzk2ZGNlOWU0OWVhNzQ3YWE3Yzg5OTU5MGQ2NzNiYWFhZmE3MjgwZDljJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Z2OW1_Eetg1JvEyvm2T1cpb61xtY0O_mSm9_1g0DmzY)
> 
> It is confusing me a lot.

We all talked about the version of onnx. You just tried to uninstall onnxruntime and upgraded onnx, check ur package's name, u need to downgraded onnx instead of onnxruntime.

**Comment by 0utsider0** [comment]
Yes ok... understood... now i fixed. 

In my defence I will tell that yesterday I was working on reactor node with version 1.16.2 all day till it broke at night with no reason. (now I know the reason, but it was working on that version 100%)

**Comment by cjvolzka** [comment]
I wasn't able to replicate the issue but I only have access to a Mac and the issue appears Windows specific. The cause isn't obvious (to me) based just on the commits cherry-picked into 1.16.2 from the previous 1.16.1 either.

Since I can't replicate it locally, I won't be able to dig into it myself. I'll continue following the thread so I can spin a 1.16.3 if needed once we know of a fix.

@liqunfu, @justinchuby, @xadupre, or @gramalingam do you know who could take this investigation on?

**Comment by liqunfu** [comment]
> I wasn't able to replicate the issue but I only have access to a Mac and the issue appears Windows specific. The cause isn't obvious (to me) based just on the commits cherry-picked into 1.16.2 from the previous 1.16.1 either.
> 
> Since I can't replicate it locally, I won't be able to dig into it myself. I'll continue following the thread so I can spin a 1.16.3 if needed once we know of a fix.
> 
> @liqunfu, @justinchuby, @xadupre, or @gramalingam do you know who could take this investigation on?

I am able to reproduce it and investigating

**Comment by mwindowshz** [comment]
> > I wasn't able to replicate the issue but I only have access to a Mac and the issue appears Windows specific. The cause isn't obvious (to me) based just on the commits cherry-picked into 1.16.2 from the previous 1.16.1 either.
> > Since I can't replicate it locally, I won't be able to dig into it myself. I'll continue following the thread so I can spin a 1.16.3 if needed once we know of a fix.
> > @liqunfu, @justinchuby, @xadupre, or @gramalingam do you know who could take this investigation on?
> 
> I am able to reproduce it and investigating. Anyone urgently needs 1.16.2 on Windows, please do: conda install conda-forge::vs2015_runtime to work around the issue. Otherwise please continue use 1.16.1 for now.

This worked!
`conda install conda-forge::vs2015_runtime`
I am using windows 1.161.1 
I used `pip install onnxruntime==1.16.1` maybe it does not bring the vc2015 runtime
at first had 1.18.1 and did not work

maybe all that is missing is the vs2015 runtime  needs to be checked

Thanks!

**Comment by Hungarogame** [comment]
Hey! I encountered the same problem while using comyfui, however I don't know how to solve it.
![image](https://github.com/user-attachments/assets/fed2b505-32b1-48e2-8b4d-671dc205c5ff)

**Comment by HANDSOME122** [comment]
![Snipaste_2024-08-14_10-46-09](https://github.com/user-attachments/assets/d283bafe-0820-4e4a-9c59-8838b286894f)
Hello, I also encountered such an error problem when using comfyui. I am not a computer major. What should I do? I saw comments saying that I need to downgrade onnx. What should I do?

**Comment by tomid99999** [comment]
> ![Snipaste_2024-08-14_10-46-09](https://private-user-images.githubusercontent.com/59920903/357661063-d283bafe-0820-4e4a-9c59-8838b286894f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM2NDA5NzksIm5iZiI6MTcyMzY0MDY3OSwicGF0aCI6Ii81OTkyMDkwMy8zNTc2NjEwNjMtZDI4M2JhZmUtMDgyMC00ZTRhLTljNTktODgzOGIyODY4OTRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE0VDEzMDQzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2OGNlNzAzNzNkY2M4YWZhNzhlOGI2ZGM5NGEwMzMwZWJlMGU4MjI5NDU2ODY0ZGFiMDU5YjA0NTk2YzIyNDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.v0hOFtJ2n5CJA31iWneEg6akPS7AhYzKw4g9TR2g5dc) Hello, I also encountered such an error problem when using comfyui. I am not a computer major. I saw comments saying that I need to downgrade onnx. What should I do?

Just downgrade the onnx version using 
`pip install onnx==1.16.1` 
and you should be good to go

**Comment by andife** [comment]
@liqunfu Could / Should we use your workaround (https://github.com/onnx/onnx/issues/6267#issuecomment-2269555846) to create a 1.16.3 ?

**Comment by IssamAbdoh** [comment]
ImportError                               Traceback (most recent call last)
Cell In[10], line 6
      4 from sklearn.model_selection import GridSearchCV
      5 import joblib
----> 6 from skl2onnx import convert_sklearn
      7 from skl2onnx.common.data_types import FloatTensorType
      8 import onnx

File [c:\Users\essam\AppData\Local\Programs\Python\Python311\Lib\site-packages\skl2onnx\__init__.py:15](file:///C:/Users/essam/AppData/Local/Programs/Python/Python311/Lib/site-packages/skl2onnx/__init__.py:15)
     11 __model_version__ = 0
     12 __max_supported_opset__ = 21  # Converters are tested up to this version.
---> 15 from .convert import convert_sklearn, to_onnx, wrap_as_onnx_mixin  # noqa
     16 from ._supported_operators import update_registered_converter, get_model_alias  # noqa
     17 from ._parse import update_registered_parser  # noqa

File [c:\Users\essam\AppData\Local\Programs\Python\Python311\Lib\site-packages\skl2onnx\convert.py:5](file:///C:/Users/essam/AppData/Local/Programs/Python/Python311/Lib/site-packages/skl2onnx/convert.py:5)
      3 import warnings
      4 from uuid import uuid4
----> 5 from .proto import get_latest_tested_opset_version
      6 from .common._topology import convert_topology
      7 from .common.utils_sklearn import _process_options

File [c:\Users\essam\AppData\Local\Programs\Python\Python311\Lib\site-packages\skl2onnx\proto\__init__.py:8](file:///C:/Users/essam/AppData/Local/Programs/Python/Python311/Lib/site-packages/skl2onnx/proto/__init__.py:8)
      1 # SPDX-License-Identifier: Apache-2.0
...
   (...)
    111     Version,
    112 )

ImportError: DLL load failed while importing onnx_cpp2py_export: A dynamic link library (DLL) initialization routine failed.

**Comment by IssamAbdoh** [comment]
Downgrading to 1.16.1 , did not solve the issue
pip uninstall onnxruntime onnxruntime-gpu
pip install onnxruntime==1.16.1

**Comment by IssamAbdoh** [comment]
These are the versions' combinantion that worked for me :
onnxruntime==1.16.1
onnx~=1.16.1
onnxconverter-common~=1.14.0

**Comment by figoliu** [comment]
> # Bug Report
> ### Is the issue related to model conversion?
> No. I can't even perform imports.
> 
> ### Describe the bug
> My projects are permissive with respect to which `onnx` PyPI package version is installed. `onnx 1.16.2` came out this morning and broke my projects.
> 
> For example, in a turnkeyml environment that used to work:
> 
> ```
> pip install --upgrade onnx
> turnkey -h
> ```
> 
> results in:
> 
> ```
> (tkml) PS C:\work\turnkeyml> turnkey -h 
> Traceback (most recent call last):
>   File "C:\Users\jefowers\AppData\Local\miniconda3\envs\tkml\lib\runpy.py", line 194, in _run_module_as_main  
>     return _run_code(code, main_globals, None,
>   File "C:\Users\jefowers\AppData\Local\miniconda3\envs\tkml\lib\runpy.py", line 87, in _run_code
>     exec(code, run_globals)
>   File "C:\Users\jefowers\AppData\Local\miniconda3\envs\tkml\Scripts\turnkey.exe\__main__.py", line 4, in <module>
>   File "C:\work\turnkeyml\src\turnkeyml\__init__.py", line 3, in <module>
>     from .files_api import evaluate_files
>   File "C:\work\turnkeyml\src\turnkeyml\files_api.py", line 8, in <module>
>     from turnkeyml.sequence import Sequence
>   File "C:\work\turnkeyml\src\turnkeyml\sequence\__init__.py", line 1, in <module>
>     from .sequence import Sequence
>   File "C:\work\turnkeyml\src\turnkeyml\sequence\sequence.py", line 11, in <module>
>     import turnkeyml.common.status as status
>   File "C:\work\turnkeyml\src\turnkeyml\common\status.py", line 10, in <module>
>     import turnkeyml.common.analyze_model as analyze_model
>   File "C:\work\turnkeyml\src\turnkeyml\common\analyze_model.py", line 4, in <module>
>     import onnx
>   File "C:\Users\jefowers\AppData\Local\miniconda3\envs\tkml\lib\site-packages\onnx\__init__.py", line 77, in <module>
>     from onnx.onnx_cpp2py_export import ONNX_ML
> ImportError: DLL load failed while importing onnx_cpp2py_export: A dynamic link library (DLL) initialization routine failed.
> ```
> 
> Setting `onnx<1.16.2` resolves the issue.
> 
> ### System information
> * OS Platform and Distribution (_e.g. Linux Ubuntu 20.04_): Windows 11
> * ONNX version (_e.g. 1.13_): 1.16.2
> * Python version: 3.10 (works on Python 3.8)
> * Protobuf version:  3.20.2
> 
> ### Reproduction instructions
> ```
> conda create -n otest python=3.10
> conda activate otest
> pip install turnkeyml==3.0.1
> turnkey -h
> ```
> 
> ### Expected behavior
> Patch version increases to packages (1.16.1 -> 1.16.2) should not include breaking changes.

winget install --id Microsoft.VCRedist.2015+.x64 --source winget

**Comment by Buumcode** [comment]
Today ComfyUI ReActor Node show no loading error: **ImportError: DLL load failed while importing onnx_cpp2py_export: A dynamic link library (DLL) initialization routine failed.** (But it works well around one month on onnx 1.16.2 before that night).

Thanx to comments above, it suggest to downgrade ONNX (not onnxRUNTIME, but ONNX) from 1.16.2 to 1.16.1.
I make it -- and now all things works well.

I've using this three commands in command-prompt. 
First is showing version what you have now. 
Second uninstall. 
Third is install 1.16.1.

1) python -m pip show onnx
2) python -m pip uninstall onnx
3) python -m pip install onnx==1.16.1

**Comment by joshuacwnewton** [comment]
This issue is still present for `onnx==1.17.0`. Limiting to `onnx<1.16.2` is the correct workaround for now.

((Also, to the reports above: This is an issue with **onnx**, not onnxruntime. Limiting the version of onnxruntime will have no effect on this. `onnx==1.16.1` + `onnxruntime==1.19.2` (latest) is a working combination.))

**Comment by andife** [comment]
@liqunfu

**Comment by gramalingam** [comment]
Can you try `conda install conda-forge::vs2015_runtime` as Liqun describes earlier in the thread?

**Comment by joshuacwnewton** [comment]
> Can you try `conda install conda-forge::vs2015_runtime` as Liqun describes earlier in the thread?

The project I was testing this for is a pip package (see https://github.com/ivadomed/ivadomed/pull/1318), and the [GitHub Actions workflow](https://github.com/ivadomed/ivadomed/blob/46248ae8206174c4b303a0cadad88754330019f0/.github/workflows/run_tests.yml#L77-L113) for that project does not use conda. 

Sample failing CI run for reference: https://github.com/ivadomed/ivadomed/actions/runs/11180519695/job/31082433890#step:7:704

Use of 1.16.2/1.17.0 is not urgent for us, though, so pinning to <1.16.2 is fine for our needs. :)

**Comment by carlo711** [comment]
can confirm too, that onnx==1.16.1 and onnxruntime==1.19.2 just works fine on windows.

**Comment by titaiwangms** [comment]
Hi @jeremyfowers,

I can't repro this on conda anymore, would you be able to repro it again? I think conda python now includes vs2015_runtime by default now.

**Comment by titaiwangms** [comment]
Hi @joshuacwnewton Do you still repro this on CI?

**Comment by joshuacwnewton** [comment]
Thank you for opening the PR on our project to test this. I am still seeing errors for [Python 3.8 CI](https://github.com/ivadomed/ivadomed/actions/runs/11244506304/job/31370891094?pr=1323#step:7:696), though given it has reached EOL this week, it is probably not a priority for us to get Py3.8 working.

**Comment by titaiwangms** [comment]
Please make sure you have at least MSVC runtime 2015 installed. For conda users, this should be included when python is installed in the environment. For pip users, please make sure the environment supports it. On Github CI, Windows platform with python installed supports MSVC runtime, except py3.8.

**Comment by ramabyg** [comment]
I also faced issue with onnx 1.17. Downgraded to 1.16.1

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "c:\rama\learn\opencv_python-1\.conda\Lib\site-packages\onnx\__init__.py", line 77, in <module>
    from onnx.onnx_cpp2py_export import ONNX_ML
ImportError: DLL load failed while importing onnx_cpp2py_export: A dynamic link library (DLL) initialization routine failed.

**Comment by soumendukrg** [comment]
I also faced the same issue with 1.17. Downgrading to 1.16.1 solved the issue.

**Comment by chiehw** [comment]
> I also faced the same issue with 1.17. Downgrading to 1.16.1 solved the issue.我在 1.17 中也遇到了同样的问题。降级到 1.16.1 解决了该问题。

Thank you, I also solved it by this way.

**Comment by pvbang** [comment]
pip install onnx==1.16.1 onnxruntime==1.19.2

**Comment by phanductri** [comment]
I have the same issue.    from onnx.onnx_cpp2py_export import ONNX_ML
ImportError: DLL load failed while importing onnx_cpp2py_export: A dynamic link library (DLL) initialization routine failed.

BUT, I down the version to 1.16.1 (using pip install onnx==1.16.1 onnxruntime==1.19.2 as above) to solve it

**Comment by kmaehashi** [comment]
In my environment, the problem was resolved after updating the MSVC Runtime from 14.29.30133 to 14.40.33816.

https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170#latest-microsoft-visual-c-redistributable-version

**Comment by qhanson** [comment]
> pip install onnx==1.16.1 onnxruntime==1.19.2

This worked for me on win11 with cuda 12.4, torch2.3.1

**Comment by ahsannawazch** [comment]
> Downgrading to 1.16.1 seems to fix it for me.

thanks

**Comment by jorickert** [comment]
@gramalingam, why was this issue closed as completed? We still see issues with ONNX version 1.16.2 on Windows. 
Does this mean that MSVC Runtime version 14.40.33816 is required for ONNX 1.16.2?

**Comment by Buumcode** [comment]
[SOLVED] ImportError: DLL load failed while importing onnx_cpp2py_export: A dynamic link library (DLL) initialization routine failed.

Make sure, your CuDNN is propely installed. Install only by ZIP file (tarball)!
Download page:
https://developer.nvidia.com/cudnn-downloads
Windows > x86_64 > Tarball > 12
And place (unzip, copy) files properly in two places:
C:\Program Files\NVIDIA\CUDNN\v9.6\bin
C:\Program Files\NVIDIA\CUDNN\v9.6\include
C:\Program Files\NVIDIA\CUDNN\v9.6\lib
AND in:
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.6\bin
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.6\include
C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v12.6\lib
Accord your steps by this articles:
https://medium.com/aimonks/install-onnxruntime-on-windows-2ad5ed8f4a50
https://docs.nvidia.com/deeplearning/cudnn/latest/installation/windows.html

Only after that my error was solved.

**Comment by xieofxie** [comment]
I put import olive in the first line solved this issue, weird

**Comment by poweiw** [comment]
We still see this issue with Windows11+Python 3.10+ONNX 1.17.0+vs2019. Any updates on this?

**Comment by andife** [comment]
> We still see this issue with Windows11+Python 3.10+ONNX 1.17.0+vs2019. Any updates on this?

I assume that it will work again with the upcoming 1.18 release (the process will start shortly). 

Have you checked if it works with the current onnx-weekly? 

@ramkrishna2910

**Comment by poweiw** [comment]
@andife Is there a planned schedule for 1.18?

**Comment by andife** [comment]
> [@andife](https://github.com/andife) Is there a planned schedule for 1.18?

The release logistics should be found at https://github.com/onnx/onnx/wiki/Logistics-for-ONNX-Release-1.18.0 @ramkrishna2910

**Comment by andife** [comment]
@poweiw You can test the rc1 at https://test.pypi.org/project/onnx/1.18.0rc1/

**Comment by joshuacwnewton** [comment]
Hi! I previously encountered issues with Python 3.8 and onnx 1.16.2 (see: https://github.com/onnx/onnx/issues/6267#issuecomment-2393755161 and the [subsequent discussion](https://github.com/onnx/onnx/issues/6267#issuecomment-2405782148)). Since then, we have moved off of Python 3.8 and switched to Python 3.9, and onnx 1.17.0 has been working for us. 🎉 

However, surprisingly, it seems that the Windows-specific issues have come back for us due to the May 12, 2025 release of [`onnx==1.18.0`](https://pypi.org/project/onnx/1.18.0/): 

- https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4900

From that issue:

> - May 12, 2025 (passing): https://github.com/spinalcordtoolbox/spinalcordtoolbox/actions/runs/14970539402/job/42050278905
> - May 13, 2025 (failing): https://github.com/spinalcordtoolbox/spinalcordtoolbox/actions/runs/14994949475
> 
> Diff between dependency versions: https://www.diffchecker.com/7cD5tLSF/

I will limit to `onnx<1.18.0` for now for Python 3.9. 

Note: Maybe 1.18.0 works for Python 3.10+? (Seems to be a repeat of the Python 3.8/3.9 story we encountered last year, where only the oldest Python version is encountering issues...)

**Comment by andife** [comment]
@joshuacwnewton could you share the error log?

**Comment by joshuacwnewton** [comment]
> @joshuacwnewton could you share the error log?

Unfortunately there were no log messages. The CI runs seem to stop abruptly, see the linked runs in 
https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4900#issue-3060671933.

**Comment by joshuacwnewton** [comment]
Update: I think the issue I reported above may be specific to the `monai` package, rather than `onnx`-related. Importing `onnx` alone works OK for 1.18.0.

Please feel free to close this issue, and I will report my issues downstream to `monai`.

**Comment by skywall** [comment]
If you're using choco, you can resolve this by installing [vcredist140](https://community.chocolatey.org/packages/vcredist140):
```bash
choco install vcredist140
```

**Comment by jgerityneurala** [comment]
I tangled with this bug on Win10 recently, and was able to resolve it by upgrading the MSVC Redistributable. In my observation: **upgrading to MSVC Redist 14.38 or higher seems to resolve this bug** on every system I was able to reproduce it on.

I did some investigation using multiple releases of the redist to try and find out what the _minimum_ version that fixes the issue was. 

Here were my observations from doing a manual bisect over [all available MSVC Redist versions](https://github.com/abbodi1406/vcredist/tree/9575c9e8745aa3fd8842cdf3e82064a3908f8523/source_links), starting from the known-bad version I had locally (`14.34.31931.0`), knowing that upgrading to MSVC redist `14.38.33130.0` had resolved the issue (in part thanks to multiple reports in this thread that `14.40.33816` resolved the problem, thank you all 💙)

```
BAD (start) 14.34.31931.0
BAD 14.36.32530
BAD 14.36.32547
OK 14.38.32919.0
OK (start) 14.38.33130.0
```

Since the way the MSVC Redist works on user systems is fairly strange (installing a version may not affect the `msvcp140.dll` that will be resolved in `System32`) I'll also share the little testing harness I used to reproduce the bug using a copy of `msvcp140.dll` in the same directory as the test program:

```python
import ctypes
# manually load the target MSVC Redist, prevents loading a newer copy; slightly evil, not OK for production, but does reproduce the issue if MSVC version is bad
msvc = ctypes.cdll.LoadLibrary("./msvcp140.dll")

# trigger the DLL initialization failure, if present
from onnx.onnx_cpp2py_export import ONNX_ML
```

If you're stuck on an old `onnx` that is affected by this very frustrating bug, I hope this information helps.

**Comment by oising** [comment]
I'm on win 11 **arm64** (surface laptop 7 with npu), and the VC redist latest versions (arm64 flavor of course) installed from winget (as of today) don't seem to drop `msvcp140.dll` at all. There's `mscp140d.dll` and myriad other suffixes in system32, but the one that onnx wants, plain old msvcp140.dll is not there. I did, however, find it in the following places:

```
C:\Windows>dir /s /b msvcp140.dll
C:\Windows\System32\HealthAttestationClient\msvcp140.dll
C:\Windows\System32\Microsoft-Edge-WebView\msvcp140.dll
C:\Windows\SysWOW64\msvcp140.dll
C:\Windows\WinSxS\arm64_microsoft-edge-webview_31bf3856ad364e35_10.0.26100.1_none_8fe1cb90f34a0e66\msvcp140.dll
C:\Windows\WinSxS\arm64_microsoft-edge-webview_31bf3856ad364e35_10.0.26100.7623_none_2e9c2a393ad2d7e1\msvcp140.dll
C:\Windows\WinSxS\arm64_microsoft-edge-webview_31bf3856ad364e35_10.0.26100.7824_none_2e8673993ae32964\msvcp140.dll
C:\Windows\WinSxS\arm64_microsoft-edge-webview_31bf3856ad364e35_10.0.26100.8036_none_2eddc7993aa1939c\msvcp140.dll
C:\Windows\WinSxS\arm64_microsoft-windows-healthattestation-csp_31bf3856ad364e35_10.0.26100.7309_none_c6e5e8d76b5545bc\msvcp140.dll
```

So I grabbed the first one and placed in `.venv\lib\site-packages\onnx` where it was being looked for, and then everything started working. Very, very annoying.

UPDATE: @jgerityneurala has got it right -- I went to his direct listing of redist downloads and grabbed 14.50+ and _now_ I see msvcp140.dll in system32. I was able to remove the version I dumped straight into my venv and it was picked up via the normal library resolution process. Thanks!
